src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1308,6 +1308,48 @@ bool QgsOgrProvider::deleteAttributes( const QgsAttributeIds &attributes )
 #endif
 }
 
+bool QgsOgrProvider::renameAttributes( const QgsFieldNameMap& renamedAttributes )
+{
+  if ( !doInitialActionsForEdition() )
+    return false;
+
+#if defined(GDAL_VERSION_NUM) && GDAL_VERSION_NUM >= 1900
+  QgsFieldNameMap::const_iterator renameIt = renamedAttributes.constBegin();
+  bool result = true;
+  for ( ; renameIt != renamedAttributes.constEnd(); ++renameIt )
+  {
+    int fieldIndex = renameIt.key();
+    if ( fieldIndex < 0 || fieldIndex >= mAttributeFields.count() )
```

Comment:
> Hmm... does QgsOgrProvider::deleteAttributes look like it's missing handling for this to you?

Indeed. Fixed and tested per fb49f700f28c1104a6c09048a10ee4df945bdc15 (couldn't test with the GPKG driver since it lacks DeleteField support. Moved test to SQLite driver)